src/components/calcite-input/calcite-input.tsx

```diff
@@ -0,0 +1,366 @@
+    | "vertical"
+    | "horizontal"
+    | "none" = "vertical";
+
+  /** specify the alignment of dropdown, defaults to left */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** is the input required */
+  @Prop() required: boolean = false;
+
+  /** should the input autofocus */
+  @Prop() autofocus: boolean = false;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // validate props
+    let statusOptions = ["invalid", "valid", "idle"];
+    if (!statusOptions.includes(this.status))
+      this.status = getElementProp(this.el, "status", "idle");
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+
+    let scale = ["s", "m", "l"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+
+    let width = ["auto", "half", "full"];
+    if (!width.includes(this.width)) this.width = "full";
+
+    let appearance = ["minimal", "default"];
+    if (!appearance.includes(this.appearance)) this.appearance = "default";
+
+    let numberButtonType = ["vertical", "horizontal", "none"];
+    if (
+      this.inputType === "number" &&
+      !numberButtonType.includes(this.numberButtonType)
+    )
+      this.numberButtonType = "vertical";
+  }
+
+  componentDidLoad() {
+    this.childEl = this.el.querySelector(`${this.childElType}`);
+  }
+
+  componentWillLoad() {
+    this.inputType = this.el.getAttribute("type");
+    this.inputValue = this.el.getAttribute("value");
+    this.childElType = this.inputType === "textarea" ? "textarea" : "input";
+    if (!this.icon && this.iconTypeDefaults[this.inputType])
+      this.icon = this.iconTypeDefaults[this.inputType];
+  }
+
+  componentWillUpdate() {
+    this.calciteInputChange.emit({
+      element: this.childEl as HTMLInputElement,
+      value: this.inputValue
+    });
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    const icon = this.setIcon(this.icon);
+    const attributes = this.getAttributes();
+    const loader = (
+      <div class="calcite-input-loading">
+        <calcite-progress type="indeterminate"></calcite-progress>
+      </div>
+    );
+
+    const numberButtonClassModifier =
+      this.numberButtonType === "horizontal"
+        ? "number-button-item-horizontal"
+        : null;
+
+    const numberButtonsHorizontalUp = (
+      <div
+        class={`calcite-input-number-button-item ${numberButtonClassModifier}`}
+        onClick={e => this.handleNumberButtonClick(e)}
+        data-adjustment="up"
+      >
+        <calcite-icon icon="chevron-up" filled></calcite-icon>
+      </div>
+    );
+
+    const numberButtonsHorizontalDown = (
+      <div
+        class={`calcite-input-number-button-item ${numberButtonClassModifier}`}
+        onClick={e => this.handleNumberButtonClick(e)}
```

Comment:
I believe this will generate a new function on each render. You could avoid it by rewriting the following way:

```tsx
 <div {/*...*/}
      onClick={this.handleNumberButtonClick}
      {/*...*/}
 >
 </div>

// ... 

private handleNumberButtonClick = (e) => {
  // ...
}
```

Reply:
Yes, good catch ⚾️